dev/source/docs/ek3-affinity-lane-switching.rst

```diff
@@ -0,0 +1,68 @@
+.. _ek3-affinity-lane-switching:
+
+================================
+EKF3 Affinity and Lane Switching
+================================
+
+The :ref:`EKF <dev:extended-kalman-filter>` instantiates multiple instances of the filter called 'lanes'. The primary lane is the one that provides state estimates, rest are updated in the background and available for switching to, in case. The number of lanes is exclusively equal to the number of IMUs enabled for use. Conventionally, each lane uses the primary instance of the Airspeed, Barometer, GPS and Magnetometer sensors. The primary sensor can be set as a user-modifiable parameter, but can be later changed by the system, even in-flight, in case of a driver-level fault. However, modern-day vehicles are known to have multiple sensors installed of good quality. Affinity is a way for the EKF lanes to use non-primary sensors with the running lanes. This provides a statistically consistent way to make use of multiple high quality sensors and use lane-switching to select the lane which has best performing combination of sensors. The lane error score takes into account innovations from all sensors affinity is available for. This way, the vehicle can be saved from mishaps using noisy non-IMU sensors as well.
```

Comment:
"in case" of what?
Suggestion: "The number of possible lanes is exactly equal to..."
Suggestion:"Affinity is a way for the EKF lanes to use non-primary sensors within any running lane"
Suggestion:"The lane error score takes into account innovations from all sensors used by a lane"

Reply:
thanks, I've modified according to your suggestions